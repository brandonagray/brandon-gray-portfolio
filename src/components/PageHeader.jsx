import React from 'react';
import styled from 'styled-components';
import { isIOS } from 'react-device-detect';

import { mediaSize } from '../data/configOptions';

const Header = styled.div`
  display: inline-block;
  margin-bottom: 0.5em;
  font-size: 8vh;
  font-weight: bold;
  font-family: 'Lato', 'Cabin', 'Ubuntu', sans-serif;

  ${mediaSize.tablet`
    font-size: 4em;
    letter-spacing: ${isIOS ? '-0.05em' : 0};
    margin-bottom: 1em;
  `} ${mediaSize.phone`
    font-size: 3em;
  `};
`;

const PageHeader = props => <Header>{props.title}</Header>;

export default PageHeader;
