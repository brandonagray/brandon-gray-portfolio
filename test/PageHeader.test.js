import React from 'react';
import PageHeader from '../src/components/PageHeader';

describe('Page Header', () => {
  jest.mock('react-device-detect', () => ({
    isIOS: false
  }));

  it('should render the header', () => {
    const header = render(
      <PageHeader className="pageHeader" title="Test Header" />
    );

    expect(header).toMatchSnapshot();
  });

  it('should have narrower letter spacing on iOS', () => {
    const header = render(
      <PageHeader className="pageHeader" title="Test Header" />
    );

    expect(header).toMatchSnapshot();
  });
});
