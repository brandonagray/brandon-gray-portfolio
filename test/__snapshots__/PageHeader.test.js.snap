// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Header should have narrower letter spacing on iOS 1`] = `
.c0 {
  display: inline-block;
  margin-bottom: 0.5em;
  font-size: 8vh;
  font-weight: bold;
  font-family: 'Lato','Cabin','Ubuntu',sans-serif;
}

@media (max-width:64em) {
  .c0 {
    font-size: 4em;
    -webkit-letter-spacing: 0;
    -moz-letter-spacing: 0;
    -ms-letter-spacing: 0;
    letter-spacing: 0;
    margin-bottom: 1em;
  }
}

@media (max-width:26.5625em) {
  .c0 {
    font-size: 3em;
  }
}

<div
  class="c0"
>
  Test Header
</div>
`;

exports[`Page Header should render the header 1`] = `
.c0 {
  display: inline-block;
  margin-bottom: 0.5em;
  font-size: 8vh;
  font-weight: bold;
  font-family: 'Lato','Cabin','Ubuntu',sans-serif;
}

@media (max-width:64em) {
  .c0 {
    font-size: 4em;
    -webkit-letter-spacing: 0;
    -moz-letter-spacing: 0;
    -ms-letter-spacing: 0;
    letter-spacing: 0;
    margin-bottom: 1em;
  }
}

@media (max-width:26.5625em) {
  .c0 {
    font-size: 3em;
  }
}

<div
  class="c0"
>
  Test Header
</div>
`;
